js/components/Slide.react.js

```diff
@@ -1,18 +1,19 @@
 import React, { Component } from 'react';
-import { connect } from 'react-redux';
-import {changeActiveSlide, changeMode} from "actions/AppActions";
+import * as Derivable from 'derivable';
+import * as AppState from 'AppState';
 
 export default class Slide extends Component {
 
   onClick() {
-    const dispatch = this.props.dispatch;
-    dispatch(changeActiveSlide(this.props.index));
-    dispatch(changeMode('full'));
+    Derivable.transact(() => {
+      AppState.changeActiveSlide(this.props.index);
+      AppState.changeMode('full');
+    });
```

Comment:
It's really a bad idea to dispatch multiple actions at once. Should refactor into `zoomSlide` action or something like this.